src/Illuminate/Config/Repository.php

```diff
@@ -226,6 +226,22 @@ public function push($key, $value)
         $this->set($key, $array);
     }
 
+    /**
+     * Merge items into an array configuration value.
+     *
+     * @param  string  $key
+     * @param  array  $value
+     * @return void
+     */
+    public function merge($key, array $value)
+    {
+        $array = $this->get($key, []);
+
+        $array = array_merge($array, $value);
+
+        $this->set($key, $array);
+    }
```

Comment:
What do you think of making it optionally recursive?
```suggestion
    public function merge($key, array $value, bool $recursive = false)
    {
        $array = $this->get($key, []);

        if ($recursive) {
            $array = array_merge_recursive($array, $value);
        } else {
            $array = array_merge($array, $value);
        }

        $this->set($key, $array);
    }
```

Reply:
Maybe, this parameter name is more intuitive:
```suggestion
    public function merge($key, array $value, bool $overwrite = false)
    {
        $array = $this->get($key, []);
        if ($overwrite) {
            $array = array_merge($array, $value);
        } else {
            $array = array_merge_recursive($array, $value);
        }
        $this->set($key, $array);
    }
```

Reply:
Of course, for the sake of simplicity, this could also be made a ternary operator, if you prefer that:
```suggestion
    public function merge($key, array $value, bool $overwrite = false)
    {
        $array = $this->get($key, []);
 
        $array = $overwrite
            ? array_merge($array, $value);
            : array_merge_recursive($array, $value);

        $this->set($key, $array);
    }
```

Reply:
The merge behaviour should probably be similar to existing mergers:
https://github.com/laravel/framework/blob/217640654f0968ac36b489bc7bf798c9e63b9282/src/Illuminate/Support/ServiceProvider.php#L157
and
https://github.com/laravel/framework/blob/217640654f0968ac36b489bc7bf798c9e63b9282/src/Illuminate/Foundation/Bootstrap/LoadConfiguration.php#L60